### DISCUSSION TITLE
speeding up uaa startup time

### DISCUSSION THREAD

**Discussion by ramonskie** [cloudfoundry/uaa#2074] [Q&A]
i'm noticing that allot of time has been spend on importing certificates when starting up uaa
has anyone ever looked at this of how we can speed this up?

as the most annoying thing to date of uaa is that it just takes an average of 5 minutes to startup..

**Comment by strehle** [comment]
I think you mean uaa bosh release ? because here we dont import certificates.
Last changes were the retry mechanism, see 
https://github.com/toabctl/uaa-release/blame/develop/jobs/uaa/templates/bin/pre-start.erb#L50-L57

means, the command /usr/sbin/update-ca-certificates can fail and therefore now we have a retry. Maybe the timeout of 180s should be reduced to less , but before without retry this long time was OK, see others

https://github.com/cloudfoundry/os-conf-release/blob/master/jobs/ca_certs/templates/pre-start.sh.erb#L31

@peterhaochen47

**Comment by peterhaochen47** [comment]
I think it's okay for reduce the retry interval if that's where it took the longest. There's a log line:
```
log "Failed to update-ca-certificates, retrying..."
```
that shows up when the retry happens. Does that show up in the slow start up?

**Comment by ramonskie** [comment]
ah as i saw the same behaviour with the release and k8 release i thought it was uaa it self
but i see now that the k8 https://github.com/cloudfoundry/uaa/blob/develop/k8s/templates/deployment.yml#L34-L92
also have some of this logic

**Reply by strehle** [reply]
and I thought it is in bosh , but ok here in k8s I simply cannot help, but if you have a PR as proposal which helps you, please propose one
